website/www/site/content/en/documentation/io/developing-io-overview.md

```diff
@@ -180,9 +180,17 @@ To create a Beam sink, we recommend that you use a `ParDo` that writes the
 received records to the data store. To develop more complex sinks (for example,
 to support data de-duplication when failures are retried by a runner), use
 `ParDo`, `GroupByKey`, and other available Beam transforms.
+Many data services are optimized to write batches of elements at a time,
+so it may make sense to group the elements into batches before writing.
+Persistant connectons can be initialized in a DoFn's `setUp` or `startBundle`
+method rather than upon the receipt of every element as well.
```

Comment:
Would using [shared.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/utils/shared.py) (which allows once-per-process initialization) be also applicable for sinks?

Reply:
I don't think any sinks to date have found a use/need for shared, but I suppose it's possible. I don't think it's common/specific enough to mention here though.